File: stack/text_edit.c
#include <stdio.h>
#include <stdlib.h>

void createFile() {
    char filename[100], ch;
    FILE *fp;

    printf("Enter filename to create: ");
    scanf("%s", filename);

    fp = fopen(filename, "w");
    if (fp == NULL) {
        printf("Error creating file!\n");
        return;
    }

    printf("Start typing your text (Press `~` on a new line to stop):\n");

    getchar(); // clear buffer
    while ((ch = getchar()) != '~') {
        fputc(ch, fp);
    }

    fclose(fp);
    printf("File saved successfully as %s\n", filename);
}

void readFile() {
    char filename[100], ch;
    FILE *fp;

    printf("Enter filename to open: ");
    scanf("%s", filename);

    fp = fopen(filename, "r");
    if (fp == NULL) {
        printf("Error opening file! File not found.\n");
        return;
    }

    printf("\n--- File Content ---\n");
    while ((ch = fgetc(fp)) != EOF) {
        putchar(ch);
    }
    printf("\n--- End of File ---\n");

    fclose(fp);
}

int main() {
    int choice;

    while (1) {
        printf("\n===== Simple Text Editor =====\n");
        printf("1. Create & Write File\n");
        printf("2. Open & Read File\n");
        printf("3. Exit\n");
        printf("Enter your choice: ");
        scanf("%d", &choice);

        switch (choice) {
            case 1: createFile(); break;
            case 2: readFile(); break;
            case 3: exit(0);
            default: printf("Invalid choice! Try again.\n");
        }
    }

    return 0;
}

File: stack/text_edit.py
def create_file():
    filename = input("Enter filename to create: ")
    with open(filename, "w") as f:
        print("Start typing your text (Enter '~' on a new line to stop):")
        while True:
            line = input()
            if line == "~":
                break
            f.write(line + "\n")
    print(f"File saved successfully as {filename}")


def read_file():
    filename = input("Enter filename to open: ")
    try:
        with open(filename, "r") as f:
            print("\n--- File Content ---")
            print(f.read(), end="")
            print("\n--- End of File ---")
    except FileNotFoundError:
        print("Error: File not found!")


def append_file():
    filename = input("Enter filename to append: ")
    try:
        with open(filename, "a") as f:
            print("Start typing text to append (Enter '~' on a new line to stop):")
            while True:
                line = input()
                if line == "~":
                    break
                f.write(line + "\n")
        print(f"Text appended successfully to {filename}")
    except FileNotFoundError:
        print("Error: File not found!")


def main():
    while True:
        print("\n===== Simple Text Editor =====")
        print("1. Create & Write File")
        print("2. Open & Read File")
        print("3. Append to File")
        print("4. Exit")
        choice = input("Enter your choice: ")

        if choice == "1":
            create_file()
        elif choice == "2":
            read_file()
        elif choice == "3":
            append_file()
        elif choice == "4":
            print("Exiting editor...")
            break
        else:
            print("Invalid choice! Try again.")


if __name__ == "__main__":
    main()